sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/ArrayTests.java

```diff
@@ -289,4 +289,122 @@ public void testElementNull() {
+                                new DBSPI32Literal(1),
+                                new DBSPI32Literal(2),
+                                new DBSPI32Literal(3))
+                ),
+                new DBSPTupleExpression(
+                        new DBSPVecLiteral(
+                                new DBSPI32Literal(5),
+                                new DBSPI32Literal(6),
+                                new DBSPI32Literal(7))
+                )
+        );
+        DBSPZSetLiteral result = new DBSPZSetLiteral(
+                new DBSPTupleExpression(
+                        new DBSPVecLiteral(
+                                new DBSPI32Literal(1),
+                                new DBSPI32Literal(2),
+                                new DBSPI32Literal(3),
+                                new DBSPI32Literal(4))
+                ),
+                new DBSPTupleExpression(
+                        new DBSPVecLiteral(
+                                new DBSPI32Literal(5),
+                                new DBSPI32Literal(6),
+                                new DBSPI32Literal(7),
+                                new DBSPI32Literal(4))
+                )
+        );
+
+        this.testQuery(ddl, "SELECT ARRAY_APPEND(val, 4) FROM ARR_TBL", new InputOutputPair(input, result));
+    }
+
+    @Test
+    public void testArrayAppendNullable() {
+        String ddl = "CREATE TABLE ARR_TBL(val INTEGER ARRAY)";
+
+        DBSPZSetLiteral input = new DBSPZSetLiteral(
+                new DBSPTupleExpression(
+                        new DBSPVecLiteral(true,
+                                new DBSPI32Literal(1),
+                                new DBSPI32Literal(2),
+                                new DBSPI32Literal(3))
+                ),
+                new DBSPTupleExpression(
+                        new DBSPVecLiteral(true,
+                                new DBSPI32Literal(5),
+                                new DBSPI32Literal(6),
+                                new DBSPI32Literal(7))
+                )
+        );
+        DBSPZSetLiteral result = new DBSPZSetLiteral(
+                new DBSPTupleExpression(
+                        new DBSPVecLiteral(true,
+                                new DBSPI32Literal(1),
+                                new DBSPI32Literal(2),
+                                new DBSPI32Literal(3),
+                                new DBSPI32Literal(4))
+                ),
+                new DBSPTupleExpression(
+                        new DBSPVecLiteral(true,
+                                new DBSPI32Literal(5),
+                                new DBSPI32Literal(6),
+                                new DBSPI32Literal(7),
+                                new DBSPI32Literal(4))
+                )
+        );
+
+        this.testQuery(ddl, "SELECT ARRAY_APPEND(val, 4) FROM ARR_TBL", new InputOutputPair(input, result));
+    }
+
+    @Test @Ignore
+    public void testArrayAppendInnerNullable() {
```

Comment:
This test fails because Calcite seems to expect a `Vec<T>` instead of `Vec<Option<T>>` for even when the input array has nullable elements. 


```rust
error[E0308]: mismatched types
    --> src/lib.rs:172:44
     |
172  |     append_to_collection_handle(&input0.0, &streams.0);
     |     ---------------------------            ^^^^^^^^^^ expected `std::option::Option<i32>`, found `i32`
     |     |
     |     arguments to this function are incorrect
     |
     = note: expected reference `&CollectionHandle<dbsp::utils::Tup1<std::option::Option<Vec<std::option::Option<i32>>>>, i64>`
                found reference `&CollectionHandle<dbsp::utils::Tup1<std::option::Option<Vec<i32>>>, i64>`
note: function defined here
    --> /home/abhizer/feldera/sql-to-dbsp-compiler/lib/sqllib/src/lib.rs:1427:8
     |
1427 | pub fn append_to_collection_handle<K>(data: &WSet<K>, handle: &CollectionHandle<K, Weight>)
     |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> src/lib.rs:174:58
    |
174 |       assert!(must_equal(&read_output_handle(&streams.1), &zset!(
    |  __________________________________________________________^
175 | |         Tup1::new(Some(vec!(
176 | |             None::<i32>, Some(1i32), Some(2i32), Some(3i32), Some(4i32), ))) => 1,
177 | |         Tup1::new(Some(vec!(
178 | |             None::<i32>, Some(5i32), Some(6i32), Some(7i32), Some(4i32), ))) => 1,
179 | |     )), r#"
    | |_____^ expected `i32`, found `std::option::Option<i32>`
    |
    = note: expected struct `OrdZSet<dbsp::utils::Tup1<std::option::Option<Vec<i32>>>, i64>`
               found struct `OrdZSet<dbsp::utils::Tup1<std::option::Option<Vec<std::option::Option<i32>>>>, {integer}>`
    = note: this error originates in the macro `zset` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0308`.
error: could not compile `temp` (lib test) due to 2 previous errors
```

Reply:
Yes, you found a new bug in Calcite: https://issues.apache.org/jira/browse/CALCITE-6275
I submitted a PR, if they don't like it we can change the parser unilaterally to do the right thing here.